### ISSUE TITLE
Use SC2059 for custom functions.

### ISSUE THREAD

**Issue by mkaniewski** [koalaman/shellcheck#2263]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2059
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

myprintf() {
        printf "%s\n" "${1}"
}

var="xyz"
myprintf "${var}"

```

#### Here's what shellcheck currently says:

No warning.

#### Here's what I wanted or expected to see:
```
In test.sh line 8:
myprintf "${var}"
       ^------^ SC2059: Don't use variables in the myprintf format string. Use myprintf '..%s..' "$foo".
```

Hi!
We use some custom functions to generate logs. Those functions use printf and acts like printf, however shellcheck doesn't treat them like printf. Therefore the warning SC2059 is not raised if a variable is used in a format string.
I don't see any option in the shellcheck to modify this behavior. Is the shellcheck code modification the only solution to achieve what we want?

Thanks.
